packages/services/src/view-service/fees.ts

```diff
@@ -4,7 +4,22 @@ import { assetIdFromBaseDenom } from '@penumbra-zone/wasm/asset';
 import { IndexedDbInterface } from '@penumbra-zone/types/indexed-db';
 
 // Attempts to extract a fee token, with priority in descending order, from the assets used
-// in the actions of the planner request.
+// in the actions of the transaction planner request (TPR).
+//
+// (1) For outputs, spends, swaps, scheduling auctions, and ICS withdrawals, the naive approach
+// of "using the first alternative token for which the user has a balance" is semantically
+// equivalent to "extract the input asset from the transaction request".
+//
+// (2) Swap claims are a special case where the claim must use the swap's fee token.
+//
+// (3) Ending and withdrawing auctions are another special case that requires a database query to
+// retrieve the relevant auction and the corresponding asset ID used in that auction. For example,
+// if a Dutch auction is scheduled with a fee paid using GM, ending and withdrawing the auction will
+// also attempt to use GM. However, if the user spends their entire GM balance after scheduling the
+// auction, they won't be able to end or withdraw it because it expects GM to pay for fees. In this
+// scenario, the fee system will use the first available alternative token balance from the
+// GAS_PRICES IndexedDB table. This allows the user to end and withdraw auctions using another asset,
```

Comment:
Why in this case are the alt fees limited to what's in the GAS_PRICES table and in the other cases, it's not?

Reply:
Consider case **(1)**: Performing outputs, spends, swaps, auction scheduling, or ICS withdrawals. These actions cannot be executed without an available balance of the input asset being traded. In that context, you can use the input asset specified in the transaction request because there is a _guarantee_ that the user has a balance for that asset ID. Therefore, the information encoded in the request is sufficient.

Consider case **(3)**: Performing auction ends, auction withdrawals, or undelegations. These are _secondary_ actions, as they require a prior action to occur, such as scheduling for auctions or delegating for undelegations. If the original assets used in those initial transactions have been spent or transferred out, retrieving the asset ID from the `GAS_PRICES` table for which the user has an available balance acts as a fallback for covering the fee. The downside is that you might have to use a different asset ID (e.g., if you auction GM --> UM, you might pay fees in GN when ending or withdrawing the auction because you've run out of GM).

Reply:
> Consider case (1): Performing outputs, spends, swaps, auction scheduling, or ICS withdrawals. These actions cannot be executed without an available balance of the input asset being traded.

This is true, but we may not have GasPrices for the input asset, in which case we would also probably want to use a fallback option

Reply:
Interesting, why would that be the case?

Reply:
@Valentine1898 specifically [getAltGasPrices](https://github.com/penumbra-zone/web/blob/main/packages/storage/src/indexed-db/index.ts#L421) should be modified to only retrieve assets for which `GasPrices` are non-zero.

Reply:
for https://github.com/penumbra-zone/web/blob/main/packages/storage/src/indexed-db/index.ts#L421-L426, we need to do something like this:

```
  async getAltGasPrices(): Promise<GasPrices[]> {
    const allGasPrices = await this.db.getAll('GAS_PRICES');
    return allGasPrices
      .map(gp => GasPrices.fromJson(gp))
      .filter(gp => 
        !gp.assetId?.equals(this.stakingTokenAssetId)
        // (gp.blockSpacePrice > 0n || gp.compactBlockSpacePrice > 0n || gp.verificationPrice > 0n || gp.executionPrice > 0n)
      ); 
  }
  ```

the issue is once we make this change and bump the idb version, nothing will work for us locally because all our alternative fee tokens has zero gas prices.

Reply:
if we don't check `(gp.blockSpacePrice > 0n || gp.compactBlockSpacePrice > 0n || gp.verificationPrice > 0n || gp.executionPrice > 0n)`, alternative fees can be paid with assets with zero gas prices.

Reply:
I moved the filtering out of the `getAltGasPrices` db call into `getAssetFromGasPriceTable`, so we don't have to perform a db change if we want to modify that function.